### PR TITLE
fix: pydantic validators and code quality

### PR DESCRIPTION
**Pull request by blenzi**
This PR fixes pydantic validators and code quality to pass mypy checks.

- dependencies added: python-dotenv to facilitate dev experience ; library stubs in quality group (optional)
- mypy config: use pydantic.mypy plugin

### CODE REVIEW

File: src/app/config.py [hunk 1]
```diff
@@ -7,6 +7,11 @@
 import secrets
 from typing import Optional
 
+from dotenv import load_dotenv
+
```
[the remark this answers is not in the document]
Comment (blenzi) [reply src/app/config.py:11]: - 👀 that doesn't seem right (it is just an import)

File: .github/workflows/style.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -r src/app/requirements.txt --upgrade
-          pip install "mypy==1.4.1" types-requests types-python-dateutil
+          pip install "mypy==1.4.1" types-requests types-python-dateutil types-passlib types-python-jose sqlalchemy-stubs
```
Comment (fe51) [inline .github/workflows/style.yml:56]: Does this mean that these libraries were missing to pass the tests  ? So it solves the current issues within the CI ?

**Review by fe51 — CHANGES_REQUESTED** [review]
Hi Bruno, 
Thanks a lot for your PR ! 

I've added a few comments on the form, and a few questions to help me understand some of the additions.  Once answered , I think it will be time to approve this PR and merge it :)

File: src/app/api/crud/authorizations.py [hunk 3]
```diff
@@ -3,6 +3,10 @@
 # This program is licensed under the Apache License 2.0.
 # See LICENSE or go to <https://opensource.org/licenses/Apache-2.0> for full license details.
 
+from typing import Optional
+
+# This program is licensed under the Apache License 2.0.
```
[the remark this answers is not in the document]
Comment (blenzi) [reply src/app/api/crud/authorizations.py]: Thanks for spotting, removed!

File: src/app/schemas/devices.py [hunk 4]
```diff
@@ -111,6 +109,5 @@ class DeviceUpdate(Login):
         description="the unique version of the current software being run on the device",
     )
 
-    @validator("last_ping", pre=True, always=True)
-    def validate_last_ping(v):
-        return None if v is None else (isoparse(v) if isinstance(v, str) else v).replace(tzinfo=None)
+    # validators
```
[the remark this answers is not in the document]
Comment (blenzi) [reply src/app/schemas/devices.py]: Indeed... it was copied from pydantic example. Removed

File: src/app/schemas/sites.py [hunk 5]
```diff
@@ -3,6 +3,10 @@
 # This program is licensed under the Apache License 2.0.
 # See LICENSE or go to <https://opensource.org/licenses/Apache-2.0> for full license details.
 
+from typing import Optional
+
+# This program is licensed under the Apache License 2.0.
```
[the remark this answers is not in the document]
Comment (blenzi) [reply src/app/schemas/sites.py]: Thanks for spotting, removed!

File: .github/workflows/style.yml (hunk 2 above)
Comment (blenzi) [reply .github/workflows/style.yml:56]: Yes, plus the other changes in the code

**Review by fe51 — APPROVED** [review]
Thanks a lot for the quick changes ! Looks good to me !

**Pull request merged**